cupy/core/core.pyx

```diff
@@ -2287,10 +2210,127 @@ cpdef _scatter_op_single(ndarray a, ndarray indices, v, int axis=0, op=''):
     if op == 'update':
         _scatter_update_kernel(
             v, indices, cdim, rdim, adim, a.reduced_view())
+    elif op == 'add':
+        if not issubclass(v.dtype.type, (numpy.int32, numpy.float32)):
+            raise TypeError(
+                'scatter_add only supports int32 and float32 as data type')
+        _scatter_add_kernel(
+            v, indices, cdim, rdim, adim, a.reduced_view())
```

Comment:
This is because those types are not supported by CUDA's atomicAdd. http://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#atomicadd
Since CuPy targets CUDA 7.5 and above, I removed support for double type even though it is supported by Pascal Architecture.

By the way, I added support for uint32 and uint64 that can be supported, but I forgot to.